src/pages/index.js

```diff
@@ -5,96 +5,235 @@ import PopupWithImage from "../components/PopupWithImage.js";
-const addNewPlaceForm = document.forms["add-place-edit-fields"];
-
 const addFormValidator = new FormValidator(config, addNewPlaceForm);
 addFormValidator.enableValidation();
 
+const avatarFormValidator = new FormValidator(config, editAvatarForm);
+avatarFormValidator.enableValidation();
+
+//Open Modals
+
 function handleImageClick(data) {
   cardPreview.open(data);
 }
 
-//Button consts
-const profileAddButton = document.querySelector(".profile__add-button");
-const profileEditButton = document.querySelector(".profile__edit-button");
-
-// button event listeners
-
 profileAddButton.addEventListener("click", () => {
   addPlacePopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+  avatarFormValidator.disableButton();
+});
+
 profileEditButton.addEventListener("click", () => {
   profileEditPopup.open();
   editFormValidator.disableButton();
-  const currentUserInfo = userInfo.getUserInfo();
-  document.querySelector("#profile-name").value = currentUserInfo.name;
-  document.querySelector("#profile-description").value =
-    currentUserInfo.description;
+  api.getUserInfo().then((res) => {
+    document.querySelector("#profile-name").value = res.name;
+    document.querySelector("#profile-description").value = res.about;
+  });
 });
 
-// form submissions
+//feature functions
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteModal,
+    handleLikeCard
+  );
   const cardElement = card.getCardInfo();
   return cardElement;
 }
 
+function handleDeleteModal(card) {
+  deletePopup.setSubmitHandler(() => handleDeleteCard(card));
+  deletePopup.open();
+}
+
+function handleDeleteCard(card) {
+  api
+    .deleteCard(card.getCardId())
+    .then(() => {
+      card.handleDeleteCard();
+      deletePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    api
+      .unlikeCard(card.getCardId())
+      .then((res) => {
+        card.handleLikeCard(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.getCardId())
+      .then((res) => {
+        card.handleLikeCard(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+//all form submissions
+
 function handleAddCardFormSubmit(data) {
-  const newCard = createCard(data);
-  cardSection.addItem(newCard);
-  addPlacePopup.close();
+  addPlacePopup.renderloading(true);
+  api
+    .addCard(data)
+    .then((data) => {
+      const newCardElement = createCard(data);
+      cardSection.addItem(newCardElement);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addFormValidator.disableButton();
+      addPlacePopup.renderloading(false);
+      addPlacePopup.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**